### PR TITLE
first draft security overview

### PR DESCRIPTION
**Pull request by rianfowler**
## Description
- draft overview of `vets-website` security requirements and existing features 
- not an expert at security so this may need significant revision
- I left off Cross-Origin Read Blocking ([CORB](https://chromium.googlesource.com/chromium/src/+/master/services/network/cross_origin_read_blocking_explainer.md#What-attacks-does-CORB-mitigate)) - AFAIK there's nothing for us to do to leverage this security feature. It might be good for people to know what it is but at this point it seems strictly a Chrome feature.

### CODE REVIEW

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- VA.gov automatically redirects to a secure protocol if a non secure connection is attempted 
+
+## Cross-Origin Resource Sharing (CORS)
+
+- Some cross origin connections made by VA.gov must support CORS including **all** XHR connections and web font downloads. 
+- Updates to CORS headers returned by `vets-api` services or `vets-website` assets must be made through the devops team and must be reviewed by the [frontend review group](https://github.com/orgs/department-of-veterans-affairs/teams/frontend-review-group). Example:  
+  - Your application is on a VA.gov subdomain, needs to connect to `vets-api`, but is not currently listed in the allowed origins
+- More [info on cors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+
+## Content Security Policy (CS)
```
Comment (rianfowler) [inline packages/documentation/src/pages/platform/front-end-standards/security]: would it be useful to document and explain our current CSP? I think most people (including me) won't understand the _why_ and I dunno if there any comments in the devops repo.

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- VA.gov automatically redirects to a secure protocol if a non secure connection is attempted 
+
+## Cross-Origin Resource Sharing (CORS)
+
+- Some cross origin connections made by VA.gov must support CORS including **all** XHR connections and web font downloads. 
+- Updates to CORS headers returned by `vets-api` services or `vets-website` assets must be made through the devops team and must be reviewed by the [frontend review group](https://github.com/orgs/department-of-veterans-affairs/teams/frontend-review-group). Example:  
+  - Your application is on a VA.gov subdomain, needs to connect to `vets-api`, but is not currently listed in the allowed origins
+- More [info on cors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+
+## Content Security Policy (CSP)
```
Comment (rianfowler) [inline packages/documentation/src/pages/platform/front-end-standards/security]: would it be useful to document and explain our current CSP? I think most people (including me) won't understand the _why_ enough to know when or why to update it

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Front end security 
+
+`vets-website` uses modern browser features to provide a secure experience for veterans. In general, these features are automatically included through the deployment process and require no configuration by front end engineers. 
+
+_This is an overview not an implementation guide. These rules apply to VA.gov production environments. Not all security features will be used in local, development, or staging environments._ 
+
+## HTTPS
```
Comment (rianfowler) [inline packages/documentation/src/pages/platform/front-end-standards/security]: maybe this should be TLS?

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+## Content Security Policy (CSP)
+
+- All VA.gov pages must return a Content Security Policy header that allows only known scripts to run on the page
+- The CSP must be *enforced* i.e. not set to `report-only`
+- The `report-uri` should point to a VA platform error capturing service (Sentry)
+  - VA.gov throttles these reports by including the `report-uri` only on a small % of responses
+- Updates to CSP headers returned by or `vets-website` assets must be made through the devops team and must be reviewed by the [frontend review group](https://github.com/orgs/department-of-veterans-affairs/teams/frontend-review-group)
+- More [info on CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+
```
Comment (rianfowler) [inline packages/documentation/src/pages/platform/front-end-standards/security]: should we include something about how staging.va.gov is gated? e.g. all test environment should at least require simple password access (dunno what terminology to use here).

**Comment by rianfowler** [comment]
I want to use this as a baseline for the onboarding documentation referenced in the attached ticket.

File: packages/documentation/src/pages/platform/front-end-standards/security (hunk 4 above)
Comment (cvalarida) [reply packages/documentation/src/pages/platform/front-end-standards/security]: It's not gated right now, yeah? Once upon a time, we had HTTP basic auth, but that was removed a while ago.

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Front end security
```
Comment (gunsch) [inline packages/documentation/src/pages/platform/front-end-standards/security]: filename: end in .md so it renders as markdown?

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Front end security 
+
+`vets-website` uses modern browser features to provide a secure experience for veterans. In general, these features are automatically included through the deployment process and require no configuration by front end engineers. 
+
+_This is an overview not an implementation guide. These rules apply to VA.gov production environments. Not all security features will be used in local, development, or staging environments._ 
+
+## HTTPS
+
+- All connections made by VA.gov must be secure (HTTPS)
```
Comment (gunsch) [inline packages/documentation/src/pages/platform/front-end-standards/security]:
```suggestion
- All connections made to VA.gov must be secure (HTTPS)
```

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+- VA.gov automatically redirects to a secure protocol if a non secure connection is attempted
```
Comment (gunsch) [inline packages/documentation/src/pages/platform/front-end-standards/security]: probably mention HSTS here too

File: packages/documentation/src/pages/platform/front-end-standards/security (hunk 2 above)
Comment (gunsch) [reply packages/documentation/src/pages/platform/front-end-standards/security]: and/or link to where our CSP rules are set?

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+- Some cross origin connections made by VA.gov must support CORS including **all** XHR connections and web font downloads. 
+- Updates to CORS headers returned by `vets-api` services or `vets-website` assets must be made through the devops team and must be reviewed by the [frontend review group](https://github.com/orgs/department-of-veterans-affairs/teams/frontend-review-group). Example:  
+  - Your application is on a VA.gov subdomain, needs to connect to `vets-api`, but is not currently listed in the allowed origins
+- More [info on cors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+
+## Content Security Policy (CSP)
+
+- All VA.gov pages must return a Content Security Policy header that allows only known scripts to run on the page
+- The CSP must be *enforced* i.e. not set to `report-only`
```
Comment (gunsch) [inline packages/documentation/src/pages/platform/front-end-standards/security]:
```suggestion
- The CSP is *enforced* (i.e. not set to `report-only`)
```

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+## Cross-Origin Resource Sharing (CORS)
+
+- Some cross origin connections made by VA.gov must support CORS including **all** XHR connections and web font downloads. 
+- Updates to CORS headers returned by `vets-api` services or `vets-website` assets must be made through the devops team and must be reviewed by the [frontend review group](https://github.com/orgs/department-of-veterans-affairs/teams/frontend-review-group). Example:  
+  - Your application is on a VA.gov subdomain, needs to connect to `vets-api`, but is not currently listed in the allowed origins
+- More [info on cors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+
+## Content Security Policy (CSP)
+
+- All VA.gov pages must return a Content Security Policy header that allows only known scripts to run on the page
```
Comment (gunsch) [inline packages/documentation/src/pages/platform/front-end-standards/security]: must, or currently do? this is automatically set by us all the time, right?

File: packages/documentation/src/pages/platform/front-end-standards/security [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+- The CSP must be *enforced* i.e. not set to `report-only`
+- The `report-uri` should point to a VA platform error capturing service (Sentry)
```
Comment (gunsch) [inline packages/documentation/src/pages/platform/front-end-standards/security]: should or does?

File: packages/documentation/src/pages/platform/front-end-standards/security (hunk 5 above)
Comment (gunsch) [inline packages/documentation/src/pages/platform/front-end-standards/security]: how is CSRF handled?

Comment (rianfowler) [reply packages/documentation/src/pages/platform/front-end-standards/security]: AFAIK we don't. we may have implemented some stuff for the direct deposit changes. I'll follow-up with @ncksllvn

**Pull request merged**